EncodingStreamingHighAvailability/HighAvailability/Services/JobOutputStatusSyncService.cs

```diff
@@ -0,0 +1,180 @@
+            {
+                foreach (var jobOutputStatusModel in jobOutputStatusModels)
+                {
+                    logger.LogInformation($"JobOutputStatusSyncService::RefreshJobOutputStatusUsingGetAsync reloading job status using API: mediaServiceInstanceName={mediaServiceAccountName} oldJobOutputStatusModel={LogHelper.FormatObjectForLog(jobOutputStatusModel)}");
+
+                    var job = await clientInstance.Jobs.GetAsync(clientConfiguration.ResourceGroup,
+                        clientConfiguration.AccountName,
+                        transformName,
+                        jobOutputStatusModel.JobName).ConfigureAwait(false);
+
+                    logger.LogInformation($"JobOutputStatusSyncService::RefreshJobOutputStatusUsingGetAsync loaded job data from API: job={LogHelper.FormatObjectForLog(job)}");
+
+                    await this.UpdateJobOutputStatusAsync(mediaServiceAccountName, jobOutputStatusModel, job, logger).ConfigureAwait(false);
+                }
+            }
+        }
+
+        private async Task UpdateJobOutputStatusAsync(string mediaServiceAccountName, JobOutputStatusModel jobOutputStatusModel, Job job, ILogger logger)
+        {
+            if (job != null)
+            {
+                var jobOutputStatusModelFromAPI = new JobOutputStatusModel
+                {
+                    Id = Guid.NewGuid().ToString(),
+                    EventTime = job.LastModified,
+                    JobOutputState = MediaServicesHelper.GetJobOutputState(job, jobOutputStatusModel.JobOutputAssetName),
+                    JobName = jobOutputStatusModel.JobName,
+                    MediaServiceAccountName = mediaServiceAccountName,
+                    JobOutputAssetName = jobOutputStatusModel.JobOutputAssetName,
+                    TransformName = jobOutputStatusModel.TransformName
+                };
+
+                await this.jobOutputStatusStorageService.CreateOrUpdateAsync(jobOutputStatusModelFromAPI, logger).ConfigureAwait(false);
+            }
+        }
+
+        private async Task RefreshJobOutputStatusUsingListAsync(string mediaServiceAccountName, IList<JobOutputStatusModel> jobOutputStatusModels, string transformName, ILogger logger)
+        {
+            var clientConfiguration = this.configService.MediaServiceInstanceConfiguration[mediaServiceAccountName];
+
+            using (var clientInstance = await MediaServicesHelper.CreateMediaServicesClientAsync(clientConfiguration).ConfigureAwait(false))
+            {
+                logger.LogInformation($"JobOutputStatusSyncService::RefreshJobOutputStatusUsingListAsync reloading job status using list API: mediaServiceInstanceName={mediaServiceAccountName} transformName={transformName}");
+                var dateFilter = DateTime.UtcNow.AddMinutes(-this.timeWindowToLoadJobsInMinutes).ToString("O", DateTimeFormatInfo.InvariantInfo);
```

Comment:
Question 4: Is this a correct way to load all jobs created since given time period? I want to make sure I am not creating inefficient query.

Reply:
Beverly confirmed that query looks good